ako-gateway-api/nodes/avi_model_l7_translator.go

```diff
@@ -40,6 +48,33 @@ func (o *AviObjectGraph) ProcessL7Routes(key string, routeModel RouteModel, pare
 	}
 }
 
+func (o *AviObjectGraph) BuildParentPGPoolHTTPPS(key string, routeModel RouteModel, parentNsName string, rules []*Rule, childVSes map[string]struct{}, fullsync bool) {
+
+	parentNode := o.GetAviEvhVS()
+	parentNs, _, parentName := lib.ExtractTypeNameNamespace(parentNsName)
+
+	// TODO(Akshay): with empty hostname at httproute, there will be no hostname attached. So need to fetch non wildcard fqdns from gw listeners
+
+	//akogatewayapiobjects.GatewayApiLister().Get
+	for _, rule := range rules {
+		// Each rule has to be converted to one httpPS
+		httpsName := akogatewayapilib.GetChildName(parentNs, parentName, routeModel.GetNamespace(), routeModel.GetName(), utils.Stringify(rule.Matches)+utils.Stringify(rule.Filters))
+		if len(rule.Filters) != 0 {
+			// build HTTPPS redirect/header modifier rules first
+			o.BuildHTTPPolicySet(key, parentNode[0], routeModel, rule, 0, httpsName)
+		}
+		if len(rule.Matches) != 0 {
+			// build HTTPPS, PG and pools
+			o.BuildParentHTTPS(key, parentNsName, parentNode[0], routeModel, rule, 0, httpsName)
+		}
+
+		if len(rule.Matches) == 0 && len(rule.Backends) != 0 {
+			// Default PG. Empty match name
+			o.BuildDefaultPGPoolForParentVS(key, parentNsName, "", parentNode[0], routeModel, rule)
+		}
```

Comment:
HTPPRoute
1. Can have only filter --> this will be handled by BuildHTTPPolicySet
2. Can have matches and backeend --> handled by o.BuildParentHTTPS
3. Can have filter and backend -->  so filter will be handled by BuildHTTPPolicySet and backend will be handled by o.BuildDefaultPGPoolForParentVS
4. Can have backend: Default PG with o.BuildDefaultPGPoolForParentVS
5. Empty backend: i couldn't think of use case.

Reply:
Could you please add that validation?
As of today, neither K8s nor AKO has that validation in place and the route->rule does not gets rejected

Reply:
@DixitAakash: HTTPRoutes, attached to same Parent Ref, with different backend refs and no other section, implies configuration will lead to 2 default actions which is wrong configuration.